library/src/androidTest/java/org/xmtp/android/library/ClientTest.kt

```diff
@@ -331,7 +331,80 @@ class ClientTest {
                 listOf(fixtures.boClient.inboxId, fixtures.caroClient.inboxId)
             )
         }
-        assertEquals(states.first().recoveryAddress.lowercase(), fixtures.bo.walletAddress.lowercase())
-        assertEquals(states.last().recoveryAddress.lowercase(), fixtures.caro.walletAddress.lowercase())
+        assertEquals(
+            states.first().recoveryAddress.lowercase(),
+            fixtures.bo.walletAddress.lowercase()
+        )
+        assertEquals(
+            states.last().recoveryAddress.lowercase(),
+            fixtures.caro.walletAddress.lowercase()
+        )
+    }
+
+    @Test
+    fun testsSignatures() {
+        val fixtures = fixtures()
+        val signature = fixtures.alixClient.signWithInstallationKey("Testing")
+        assertEquals(fixtures.alixClient.verifySignature("Testing", signature), true)
+        assertEquals(fixtures.alixClient.verifySignature("Not Testing", signature), false)
+
+        val alixInstallationId = fixtures.alixClient.installationId
+        assertEquals(
+            fixtures.alixClient.verifySignatureWithInstallationId(
+                "Testing",
+                signature,
+                alixInstallationId
+            ), true
+        )
+        assertEquals(
+            fixtures.alixClient.verifySignatureWithInstallationId(
+                "Not Testing",
+                signature,
+                alixInstallationId
+            ), false
+        )
+        assertEquals(
+            fixtures.alixClient.verifySignatureWithInstallationId(
+                "Testing",
+                signature,
+                fixtures.boClient.installationId
+            ), false
+        )
+        assertEquals(
+            fixtures.boClient.verifySignatureWithInstallationId(
+                "Testing",
+                signature,
+                alixInstallationId
+            ), true
+        )
```

Comment:
This feels weird that it's true? Should Bo really be able to verify if a signature was made by alix's installationId?